### PR TITLE
Optional JIRA task

### PR DESCRIPTION
**Pull request by two-legs**
We have a case where there is not always a task in JIRA, so it would be great to be able to make the field with JIRA optional.

### CODE REVIEW

**Comment by polarn** [comment]
Sorry for being super slow.. May I ask why you use this module if you don't have any Jira issue? :)

**Comment by two-legs** [comment]
@polarn We have a JIRA task in most cases, but we have exceptions. These may be some very small changes that we don't create tasks for.

**Comment by polarn** [comment]
Allright understand. And you don't change the default behaviour so it should be fine! Can you rebase it and make it into one single commit? Because I see my and Julius commits in the git log of this branch?

**Comment by two-legs** [comment]
Done

**Comment by polarn** [comment]
Then I try it I get this:
```
?  (optional) (DAZ-5642)
```

And before it was like:
```
? Enter JIRA issue (DAZ-12345) (DAZ-5642)
```

Haven't had time to check the code much, do you get it working correctly when you test it? Seems like the message created is not correct...

**Comment by two-legs** [comment]
Looks like the formatter broke the code after fix... 
I fixed and added tests for dynamic questions. Also I fixed similar bug with the scope question.

File: index.js [hunk 1]
```diff
@@ -47,7 +47,12 @@ const options = {
     (process.env.CZ_MAX_LINE_WIDTH &&
       parseInt(process.env.CZ_MAX_LINE_WIDTH)) ||
     config.maxLineWidth ||
-    defaults.maxLineWidth
+    defaults.maxLineWidth,
+  jiraOptional: getEnvOrConfig(
+    process.env.CZ_JIRA_OPTIONAL,
+    config.jiraMode,
```
Comment (polarn) [inline index.js:53]: @two-legs this line and the next you need to replace `jiraMode` with `jiraOptional` 😉

**Review by polarn — CHANGES_REQUESTED** [review]
Classic copy and paste miss. 😄

File: index.js (hunk 1 above)
Comment (two-legs) [reply index.js]: Fixed

**Comment by polarn** [comment]
Thanks for contributing! Next time I'll try to be faster...

**Pull request merged**